sshfs.c

```diff
@@ -3485,7 +3485,7 @@ static int sshfs_opt_proc(void *data, const char *arg, int key,
                                  * mountpoint to later be completely
                                  * unprivileged with libfuse >= 3.3.0.
                                  */
-                                sshfs.mountpoint = arg;
+                                sshfs.mountpoint = strdup(arg);
```

Comment:
I don't think this is the right solution, it introduces a memory link instead.
Can't we simply declare sshfs.mountpoint as const?

Reply:
Good point! I don't know if `fuse_opt_parse` promises to keep the strings it passes to `sshfs_opt_proc` alive until program exit. I was unable to find anything about lifetime in the documentation so better be safe than sorry and copy the value. We could of course free the previous value. What do you prefer?

Reply:
You're right. Let's copy it and make sure that it's freed in the right places.

Reply:
Ok, thanks for confirming! I will update the pull request with some `free` ;)

Reply:
The code that assigns `sshfs.mountpoint` is inside an `if (!sshfs.mountpoint)` block. Thus it is only assigned to only once, and no memory leak can occur, and no free'ing needs to be added.